### PR TITLE
MNT drop python=3.7 support

### LINKED ISSUE #245: Drop python=3.7 support
Now that colab supports 3.8, we should drop 3.7 support.

### PR DESCRIPTION
**Pull request by adrinjalali**
Same as https://github.com/skops-dev/skops/pull/246, fixes #245 , but doesn't add 3.11. #246 will add that.

### CODE REVIEW

File: skops/utils/fixes.py [hunk 1]
```diff
@@ -20,38 +18,3 @@
     # if you're removing this, you should also remove the dependency from
     # _min_dependencies.py
     from typing_extensions import Literal  # noqa
-
```
Comment (BenjaminBossan) [inline skops/utils/fixes.py:23]: The lines above, importing `metadata` and `Literal`, can also be deleted, no? Hence the whole of `fixes.py`, hence the whole of `skops/utils/`.

File: pyproject.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [tool.black]
 line-length = 88
-target_version = ['py37', 'py38', 'py39', 'py310']
+target_version = ['py38', 'py39', 'py310', 'py311']
```
[the remark this answers is not in the document]
Comment (adrinjalali) [reply pyproject.toml:3]: we do code in py311, so I think this is better to stay.

**Review by BenjaminBossan — COMMENTED** [review]
Can we not get rid of `fixes.py` completely now?

File: pyproject.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (BenjaminBossan) [reply pyproject.toml:3]: It's unclear to me what this option really does. Is it only relevant when a new Python version changes syntax? Anyway, I'm fine with keeping it.

[the remark this answers is not in the document]
Comment (adrinjalali) [reply pyproject.toml:3]: As in, the output should be valid in all these python versions, I think.

**Review by BenjaminBossan — APPROVED** [review]
Fantastic

**Pull request merged**